docs/advanced/classes.rst

```diff
@@ -80,6 +80,11 @@ helper class that is defined as follows:
         }
     };
 
+The ``py::trampoline_self_life_support`` base class is needed to ensure that a
+``std::unique_ptr`` can safely be passed between Python and C++. It is best
+practice to always use the base class, although it may not strictly be needed
+in all use cases.
```

Comment:
Add a note that this isn't needed when using `py::class_`? 

I wonder if it would just be better to force that trampolines have this via a static_assert? After all, better a compile error than a runtime error?

Reply:
What do you think about this?

commit 4cc528d15a794cae129c01e2fcb026974484fe00

Direct link to generated html: https://pybind11--5542.org.readthedocs.build/en/5542/advanced/classes.html

Reply:
> I wonder if it would just be better to force that trampolines have this via a static_assert?

Sorry I forgot to respond to this before:

Based on
* how much trouble I got myself into with [this good deed](https://github.com/pybind/pybind11/pull/4246),
* how much fallout in global testing I used to see when making such changes (while I was still working at Google),
* and my desire to make it as likely as possible that people migrate to pybind11v3 quickly,

I much rather prefer to not add such a static_assert. In commit 4cc528d15a794cae129c01e2fcb026974484fe00 I wrote "please think twice" with all of that in mind. I hope that'll do the trick for most people.

Reply:
Fair enough. LGTM.